common/src/runMain.ts

```diff
@@ -0,0 +1,105 @@
+import { apiClient } from "./client/apiClient";
+import { Assertion } from "./client/responses/runInformationResponse";
+import { Config } from "./config";
+import { Logger } from "./log";
+import { Output } from "./output";
+import { FinishedRun, waitForRunEnd } from "./run/ongoing";
+import { StartedRun, startRun } from "./run/start";
+import { isSuccessful } from "./run/status";
+import { StateStore } from "./state";
+import { formatErrorMessage, console } from "./utils";
+
+const { red, green, bright } = console;
+
+export const runMain = async (output: Output, logger: Logger, config: Config, state: StateStore): Promise<void> => {
+  try {
+    // Do not re-run the main action when launched again later
+    state.setFinished();
+
+    const client = apiClient(config.api);
+
+    await client.checkCloudCompatibility();
+
+    const startedRun = await startRun(client, config);
+    state.setRunning(startedRun.runId); // Run started, cleanup will be needed if interrupted now
+    logStart(logger, config, startedRun);
+
+    await output.set("run_id", startedRun.runId);
+    await output.set("reports_url", startedRun.reportsUrl);
+    await output.set("runs_url", startedRun.runsUrl);
+
+    if (config.waitForRunEnd) {
+      const finishedRun = await waitForRunEnd(client, logger, startedRun);
+      state.setFinished(); // Run finished, no cleanup needed
+      logAssertionResults(logger, finishedRun.assertions);
+      logResult(logger, config, startedRun, finishedRun);
+
+      await output.set("runs_status_code", finishedRun.statusCode);
```

Comment:
Isn't this the status code of a single run?

Reply:
I'm not sure I understand what you're asking? We're launching a run, waiting for a status we consider to be a "final status", and outputting that status.

Reply:
Should it be `runs_status_code` or `run_status_code` ? Same for `runs_status_name`.

Reply:
probably not important, yes, but it's also not much of a bother 🤷  I put it in a separate pull request though: https://github.com/gatling/enterprise-action/pull/39 I'd like to do a patch release of the GitHub Action with that fix _before_ I merge the big refactoring of the project build structure, just in case.